src/setuptools_scm/__init__.py

```diff
@@ -34,22 +34,27 @@ def version_from_scm(root):
     return _version_from_entrypoint(config, "setuptools_scm.parse_scm")
 
 
-def _call_entrypoint_fn(config, fn):
+def _call_entrypoint_fn(root, config, fn):
     if function_has_arg(fn, "config"):
-        return fn(config.absolute_root, config=config)
+        return fn(root, config=config)
     else:
         warnings.warn(
             "parse functions are required to provide a named argument"
             " 'config' in the future.",
             category=PendingDeprecationWarning,
             stacklevel=2,
         )
-        return fn(config.absolute_root)
+        return fn(root)
 
 
-def _version_from_entrypoint(config, entrypoint):
-    for ep in iter_matching_entrypoints(config.absolute_root, entrypoint):
-        version = _call_entrypoint_fn(config, ep.load())
+def _version_from_entrypoint(config, entrypoint, fallback=False):
```

Comment:
please infer the entry-point from the Boolean, so the entry-point name is no longer passed to this function
if you get a good idea for the new function name from that, please apply it as well

Reply:
https://github.com/pypa/setuptools_scm/blob/master/testing/conftest.py#L66 - this needs to pass fallback_root as well

Reply:
Changed to _version_from_entrypoint**s**